batchflow/models/tf/mobilenet.py

```diff
@@ -219,8 +273,71 @@ def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=6, w
         tf.Tensor
         """
         num_filters = int(cls.num_channels(inputs, kwargs.get('data_format')) * expansion_factor * width_factor)
-        conv_filters = [num_filters, num_filters, filters]
-        x = conv_block(inputs, 'cna Cna cn', conv_filters, [1, 3, 1], name=name, strides=[1, strides, 1], **kwargs)
+        x = conv_block(inputs, 'cnaWna', [num_filters, num_filters], [1, kernel_size], strides=[1, strides],
+                       name='%s-exp' % name, **kwargs)
+
+        if se_block:
+            if not isinstance(se_block, dict):
+                se_block = dict(activation=[kwargs.get('activation', tf.nn.relu), h_sigmoid], ratio=num_filters // 4)
+            se_block = {**kwargs, **se_block}
+            x = cls.se_block(x, name='%s-se' % name, **se_block)
+        x = conv_block(x, 'cn', filters, 1, name='%s-down' % name, **kwargs)
+
         if residual:
             x = inputs + x
         return x
+
+    @classmethod
+    def head(cls, inputs, se_block=None, name='head', **kwargs):
+        """ The last network layers which produce predictions
+
+        Parameters
+        ----------
+        inputs : tf.Tensor
+            input tensor
+        name : str
+            scope name
+        se_block : dict, optional
+            params for :meth:`~TFModel.se_block`
+
+        Returns
+        -------
+        tf.Tensor
+        """
+        kwargs = cls.fill_params('head', se_block=se_block, **kwargs)
+        layout, filters, se_block = cls.pop(['layout', 'filters', 'se_block'], kwargs)
+
+        if se_block:
+            x = conv_block(inputs, 'cna', filters[0], name='%s-conv1' % name, **kwargs)
+            x = cls.se_block(x, **{**kwargs, **se_block}, name='%s-se' % name)
+            x = conv_block(x, 'vcacV', filters=filters[1:], name='%s-conv2' % name, **kwargs)
+            return x
+        return conv_block(inputs, layout=layout, filters=filters, name=name, **kwargs)
```

Comment:
If `se_block` evaluates to `True`, then `layout` is not used at all. Is that intended?

Reply:
We decided to leave this the way it is. But multiple `head`s will allow to move this into a model config.